doc/design/scheduled-tasks.md

```diff
@@ -34,21 +34,46 @@ Whenever a Version is modified, a flag `needs_recomputing` is set on the Version
 
 #### Scheduled job
 Separately, a scheduled job that runs on a configurable fairly tight interval will check for any Versions that need to be updated.
-For each of those Versions, it also checks that the Version hasn't been modified in a certain amount of time (say 5 minutes). This prevents DB locks from happening while Assets are still being uploaded to the Version.
 
-The code would look something like this:
+The full code would look something like this:
 ```
 def scheduled_task():
   versions = Version.objects.filter(needs_recomputing=True).filter(modified__lt=timezone.now() - 5 minutes)
   # SELECT ... FROM api_version WHERE api_version.needs_recomputing AND now() - api_version.modified > 5 minutes;
```

Comment:
re `5 minutes` -- made a dedicated issue #431

Reply:
If those timeouts are configurable with environment variables, they could both be set to 1 second in the integration testing environment so the test only has to wait ~2 seconds for the job to run. You also get to test the actual functionality of the system instead of using a shortcut.

Reply:
good to hear that we are on the same page .  might be worth adding in design doc defining some env var to control for time outs.